mongo_repository.go

```diff
@@ -27,41 +30,57 @@ type record struct {
 	Payload   []byte
 }
 
-type MongoRepository struct {
+type MongoRepository[T DomainObject] struct {
 	collection          *mongo.Collection
 	snapshotsCollection *mongo.Collection
 	publisher           *EventPublisher
 	snapshotsCache      *ristretto.Cache
 }
 
-func NewMongoRepository(database *mongo.Database, publisher *EventPublisher) MongoRepository {
-	cache, _ := ristretto.NewCache(&ristretto.Config{
+func NewMongoRepository[T DomainObject](database *mongo.Database, publisher *EventPublisher) (*MongoRepository[T], error) {
+	cache, err := ristretto.NewCache(&ristretto.Config{
 		NumCounters: 1e3,
 		MaxCost:     1 << 30,
 		BufferItems: 64,
 	})
-	return MongoRepository{
+	if err != nil {
+		return nil, err
+	}
+	err = MigrateMongoDB(database, "./migrations")
+	if err != nil {
+		return nil, err
+	}
+	return &MongoRepository[T]{
 		collection:          database.Collection("event_store"),
 		snapshotsCollection: database.Collection("domain_event_snapshots"),
 		publisher:           publisher,
 		snapshotsCache:      cache,
-	}
+	}, nil
 }
 
-func (r *MongoRepository) Save(object DomainObject) error {
-	eventToAdd := object.CollectUnsavedEvents()
+func (r *MongoRepository[T]) Save(object T) error {
+	events := object.CollectUnsavedEvents()
 
-	records := toRecords(eventToAdd)
-	r.collection.InsertMany(context.TODO(), records)
-
-	r.publisher.Publish(eventToAdd)
+	records := toRecords(events)
+	_, err := r.collection.InsertMany(context.TODO(), records)
```

Comment:
```suggestion
	_, err := r.collection.InsertMany(context.Background(), records)
```

Etant donné qu'il est pas prévu de passer un context au Save()